crates/routes/src/feeds.rs

```diff
@@ -402,6 +439,183 @@ fn create_reply_and_mention_items(
+        &modlog_url,
+        &format!("Added admin {}", &v.other_person.name),
+        &None,
+      ),
+      ModlogCombinedView::ModAddCommunity(v) => build_modlog_item(
+        &v.moderator,
+        &v.mod_add_community.published,
+        &modlog_url,
+        &format!(
+          "Added mod {} to /c/{}",
+          &v.other_person.name, &v.community.name
+        ),
+        &None,
+      ),
+      ModlogCombinedView::ModBan(v) => build_modlog_item(
+        &v.moderator,
+        &v.mod_ban.published,
+        &modlog_url,
+        &format!("Banned {}", &v.other_person.name),
+        &v.mod_ban.reason,
+      ),
+      ModlogCombinedView::ModBanFromCommunity(v) => build_modlog_item(
+        &v.moderator,
+        &v.mod_ban_from_community.published,
+        &modlog_url,
+        &format!(
+          "Banned {} from /c/{}",
+          &v.other_person.name, &v.community.name
+        ),
+        &v.mod_ban_from_community.reason,
+      ),
+      ModlogCombinedView::ModFeaturePost(v) => build_modlog_item(
+        &v.moderator,
+        &v.mod_feature_post.published,
+        &modlog_url,
+        &format!("Featured post {}", &v.post.name),
+        &None,
+      ),
+      ModlogCombinedView::ModChangeCommunityVisibility(v) => build_modlog_item(
+        &v.moderator,
+        &v.mod_change_community_visibility.published,
+        &modlog_url,
+        &format!(
+          "Changed /c/{} visibility to {}",
+          &v.community.name, &v.mod_change_community_visibility.visibility
+        ),
+        &v.mod_change_community_visibility.reason,
+      ),
+      ModlogCombinedView::ModLockPost(v) => build_modlog_item(
+        &v.moderator,
+        &v.mod_lock_post.published,
+        &modlog_url,
+        &format!("Locked post {}", &v.post.name),
+        &v.mod_lock_post.reason,
+      ),
+      ModlogCombinedView::ModRemoveComment(v) => build_modlog_item(
+        &v.moderator,
+        &v.mod_remove_comment.published,
+        &modlog_url,
+        &format!("Removed comment {}", &v.comment.content),
+        &v.mod_remove_comment.reason,
+      ),
+      ModlogCombinedView::ModRemoveCommunity(v) => build_modlog_item(
+        &v.moderator,
+        &v.mod_remove_community.published,
+        &modlog_url,
+        &format!("Removed community /c/{}", &v.community.name),
+        &v.mod_remove_community.reason,
+      ),
+      ModlogCombinedView::ModRemovePost(v) => build_modlog_item(
+        &v.moderator,
+        &v.mod_remove_post.published,
+        &modlog_url,
+        &format!("Removed post {}", &v.post.name),
+        &v.mod_remove_post.reason,
+      ),
+      ModlogCombinedView::ModTransferCommunity(v) => build_modlog_item(
+        &v.moderator,
+        &v.mod_transfer_community.published,
+        &modlog_url,
+        &format!(
+          "Tranferred /c/{} to /u/{}",
+          &v.community.name, &v.other_person.name
+        ),
+        &None,
+      ),
+    })
+    .collect::<LemmyResult<Vec<Item>>>()?;
```

Comment:
This is where a trait could be useful. Basically define a trait like this and implement it for all the modlog structs:
```rust
trait ModlogItem {
  fn moderator(&self) -> Option<Person>;
  fn published(&self) -> DateTime<Utc>;
  ...
}
```

And then change build_modlog_item to take  `T: ModlogItem` and call these methods. Though it only makes sense if the trait can also be used elsewhere in the code. If its only needed here then we can leave it like this.

Reply:
It's only needed here. I agree it would make sense if we needed to match on all the other modlog types elsewhere.